contracts/src/internal/HyperdriveCheckpoint.sol

```diff
@@ -89,8 +69,33 @@ abstract contract HyperdriveCheckpoint is
             return checkpoint_.vaultSharePrice;
         }
 
+        // If the checkpoint time is the latest checkpoint, we use the current
+        // vault share price. Otherwise, we use a linear search to find the
+        // closest vault share price and use that to perform the checkpoint.
+        uint256 checkpointVaultSharePrice;
+        uint256 latestCheckpoint = _latestCheckpoint();
+        if (_checkpointTime == latestCheckpoint) {
+            checkpointVaultSharePrice = _vaultSharePrice;
+        } else {
+            for (
+                uint256 time = _checkpointTime;
```

Comment:
`time` could start at `_checkpointTime + _checkpointDuration` already because we know that `_checkpoints[_checkpointTime].vaultSharePrice == 0` due to the early return above and `_checkpointTime  < latestCheckpoint ` because we're in the `else`.